mesonbuild/coredata.py

```diff
@@ -717,16 +717,20 @@ def set_default_options(self, default_options, subproject, env):
         # Some options default to environment variables if they are
         # unset, set those now. These will either be overwritten
         # below, or they won't. These should only be set on the first run.
-        if env.first_invocation:
-            p_env = os.environ.get('PKG_CONFIG_PATH')
-            if p_env:
-                # PKG_CONFIG_PATH may contain duplicates, which must be
-                # removed, else a duplicates-in-array-option warning arises.
-                pkg_config_paths = []
-                for k in p_env.split(':'):
-                    if k not in pkg_config_paths:
-                        pkg_config_paths.append(k)
-                options['pkg_config_path'] = pkg_config_paths
+        p_env = os.environ.get('PKG_CONFIG_PATH')
+        if p_env:
+            # PKG_CONFIG_PATH may contain duplicates, which must be
+            # removed, else a duplicates-in-array-option warning arises.
+            p_list = list(OrderedSet(p_env.split(':')))
+            if env.first_invocation:
+                options['pkg_config_path'] = p_list
```

Comment:
It was already like that before, but this means the environment variable takes precedence over value set with `-Dpkg_config_path`. I don't think this is right, for CFLAGS it is the opposite.

Reply:
No, this is setting the default values, if there is a command line value it will overwrite the environment variables. Just like CFLAGS. I just tested to be sure.

Reply:
oh ok then.